mono/eglib/giconv.c

```diff
@@ -254,8 +254,16 @@ g_iconv (GIConv cd, gchar **inbytes, gsize *inbytesleft,
  * Unicode encoders and decoders
  */
 
-static int
-decode_utf32be (char *inbuf, size_t inleft, gunichar *outchar)
+static FORCE_INLINE (uint32_t)
+read_uint32_endian (unsigned char *inptr, unsigned endian)
+{
+	if (endian == G_LITTLE_ENDIAN)
+		return (inptr[3] << 24) | (inptr[2] << 16) | (inptr[1] << 8) | inptr[0];
+	return (inptr[0] << 24) | (inptr[1] << 16) | (inptr[2] << 8) | inptr[3];
+}
+
+static FORCE_INLINE (int)
```

Comment:
I don't think it's good idea to force inline this one

Reply:
Intent was to factor/share source w/o changing object code, but I'm ok either way. The function has to be inlined in order to propagate the constant.

Reply:
Ah, specifically utf32, which we never even use, as opposed to utf16 which we heavily use?